### PR TITLE
Map duration and time messages

### PR DESCRIPTION
**Pull request by dirk-thomas**
Only slightly related to ros2/ros1_bridge#104.

While the bridge can already handle fields with a `Duration` or `Time` type it currently can't map the messages `std_msgs/Duration|Time` with `builtin_interfaces/Duration|Time`.

With this patch the following appears when invoking the dynamic bridge with `--print-pairs`:

```
  - 'builtin_interfaces/Duration' (ROS 2) <=> 'std_msgs/Duration' (ROS 1)
  - 'builtin_interfaces/Time' (ROS 2) <=> 'std_msgs/Time' (ROS 1)
```

And the following example works:

* ROS 1 side:
  ```
  rostopic pub /time std_msgs/Time "data:
    secs: 1 
    nsecs: 2"
  ```
* ROS 2 side:
  ```ros2 topic echo /time builtin_interfaces/Time```

### CODE REVIEW

**Comment by dirk-thomas** [comment]
Waiting for review.

File: include/ros1_bridge/convert_builtin_interfaces.hpp [hunk 1]
```diff
@@ -31,27 +31,26 @@ namespace ros1_bridge
 template<>
 void
 convert_1_to_2(
-  const ros::Duration & ros1_msg,
+  const ros::Duration & ros1_type,
```
Comment (dhood) [inline include/ros1_bridge/convert_builtin_interfaces.hpp]: This change doesn't make sense to me. Can you clarify why you changed only one of the names of this prototype? and that the names don't match what's used in the implementation at https://github.com/ros2/ros1_bridge/pull/106/files#diff-e7604d29ef87753b67dcf843c9e940caR72?

Comment (mikaelarguedas) [reply include/ros1_bridge/convert_builtin_interfaces.hpp]: these are not messages but ROS 1 types hence the change in name. The prototype you're pointing to is taking messages and not types so using the `_msg` there is appropriate

Comment (dirk-thomas) [reply include/ros1_bridge/convert_builtin_interfaces.hpp]: This patch changes the `ros1_msg` argument to `ros1_type` argument for all `convert_1_to_2` / `convert_2_to_1` functions in the header as well as the source file. The rational is that neither of the two arguments `ros::Duration` nor `ros::Time` is a message but just a plain type.

The function you are referencing is unrelated to this signature. That one is for actual message in the `builtin_interfaces` packages and therefore those arguments still use `_msg` as the argument suffix.

File: resource/get_factory.cpp.em [hunk 2]
```diff
@@ -18,8 +18,8 @@
 @{
 from ros1_bridge import camel_case_to_lower_case_underscore
 }@
-#include "ros1_bridge/factory.hpp"
 #include "ros1_bridge/builtin_interfaces_factories.hpp"
+#include "ros1_bridge/factory.hpp"
```
Comment (dirk-thomas) [inline resource/get_factory.cpp.em]: I would rather revert this change. Before it was including first the factory interface and then all specialized factories. That made sense to me. Also the include order isn't being checked at the moment since some includes are hard coded and some other are being generated. So the overall order won't pass the inter anyway (without custom logic ordering the superset).

Comment (mikaelarguedas) [reply resource/get_factory.cpp.em]: Feel free to revert it, I wanted to save an iteration but apparently I should have just commented to get feedback rather than commiting the change

Comment (dirk-thomas) [reply resource/get_factory.cpp.em]: Np, better being proactive - I appreciate it!

File: include/ros1_bridge/convert_builtin_interfaces.hpp (hunk 1 above)
Comment (dhood) [reply include/ros1_bridge/convert_builtin_interfaces.hpp]: ok, I'll keep looking at this to get my head around it myself, in the meantime the approval from @mikaelarguedas suggests it's just my misunderstanding so not a blocker :smile:

Comment (dhood) [reply include/ros1_bridge/convert_builtin_interfaces.hpp]: following up, @mikaelarguedas pointed out what I was missing, which is that these particular msgs, fundamentally, are represented differently in ROS 1 and ROS 2: time/duration isn't a message in ROS 1, it is a data type, whereas in ROS 2 time/duration has message definitions associated with them. The contents of `ros1_msg.data` at [this line](https://github.com/ros2/ros1_bridge/pull/106/files#diff-e7604d29ef87753b67dcf843c9e940caR75) is not another message.
